api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/ClaimResultTest.java

```diff
@@ -0,0 +1,115 @@
+import static uk.gov.dhsc.htbhf.eligibility.model.testhelper.CombinedIdAndEligibilityResponseTestDataFactory.anIdMatchedEligibilityConfirmedUCResponseWithAllMatches;
+
+class ClaimResultTest {
+
+    @Test
+    void shouldCreateResultWithEntitlement() {
+        Claim claim = aValidClaim();
+        VoucherEntitlement voucherEntitlement = aValidVoucherEntitlement();
+        CombinedIdentityAndEligibilityResponse identityAndEligibilityResponse = anIdMatchedEligibilityConfirmedUCResponseWithAllMatches();
+
+        ClaimResult result = ClaimResult.withEntitlement(claim, voucherEntitlement, identityAndEligibilityResponse);
+
+        assertThat(result).isNotNull();
+        assertThat(result.getClaim()).isEqualTo(claim);
+        VerificationResult verificationResult = result.getVerificationResult();
+        assertThat(verificationResult.getAddressLine1Match()).isEqualTo(identityAndEligibilityResponse.getAddressLine1Match());
+        assertThat(verificationResult.getDeathVerificationFlag()).isEqualTo(identityAndEligibilityResponse.getDeathVerificationFlag());
+        assertThat(verificationResult.getEligibilityOutcome()).isEqualTo(identityAndEligibilityResponse.getEligibilityStatus());
+        assertThat(verificationResult.getEmailAddressMatch()).isEqualTo(identityAndEligibilityResponse.getEmailAddressMatch());
+        assertThat(verificationResult.getIdentityOutcome()).isEqualTo(identityAndEligibilityResponse.getIdentityStatus());
+        assertThat(verificationResult.getMobilePhoneMatch()).isEqualTo(identityAndEligibilityResponse.getMobilePhoneMatch());
+        assertThat(verificationResult.getPostcodeMatch()).isEqualTo(identityAndEligibilityResponse.getPostcodeMatch());
+        assertThat(verificationResult.getPregnantChildDOBMatch()).isEqualTo(identityAndEligibilityResponse.getPregnantChildDOBMatch());
+        assertThat(verificationResult.getQualifyingBenefits()).isEqualTo(identityAndEligibilityResponse.getQualifyingBenefits());
+    }
+
+    @ParameterizedTest
+    @MethodSource("noChildMatches")
+    void shouldSetIsPregnantOrChildMatchedToTrueWhenPregnant(List<LocalDate> declaredChildren, List<LocalDate> registeredChildren) {
+        Claim claim = aClaimWithExpectedDeliveryDateAndChildrenDobs(LocalDate.now(), declaredChildren);
+        VoucherEntitlement voucherEntitlement = aValidVoucherEntitlement();
+        CombinedIdentityAndEligibilityResponse identityAndEligibilityResponse = anIdMatchedEligibilityConfirmedUCResponseWithAllMatches(registeredChildren);
+
+        ClaimResult result = ClaimResult.withEntitlement(claim, voucherEntitlement, identityAndEligibilityResponse);
+
+        assertThat(result.getVerificationResult().getIsPregnantOrAtLeast1ChildMatched()).isTrue();
+    }
+
+    @ParameterizedTest
+    @MethodSource("atLeastOneChildMatches")
+    void shouldSetIsPregnantOrChildMatchedToTrueWhenChildMatches(List<LocalDate> declaredChildren, List<LocalDate> registeredChildren) {
```

Comment:
Should probably be `shouldSetIsPregnantOrChildMatchedToTrueWhenChildMatchesAndNotPregnant` to be 100% clear.